Recover automatic block number calculation

xOneca opened the pull request:
Total block count was removed on commit 457078c without explanation. Bring it back with a change: instead of increasing the number of blocks specified by the user (like it used to do), we keep the current behavior of crying to the user.

Commit:
Define {MAX_,}RESERVED_BLOCKS as floats

As it was defined could not work in other contexts (the current lone use works
by chance, because multiplication is done first), where the division may yield
an unexpected (but valid) zero result.

Commit:
Recover automatic block number calculation

The total block count was once done automatically but it was lost on commit
457078c90d5a3738ee2ed2206728b1b5cacecb9f.

The calculation is recovered, but made optional: it is only done if total blocks
is not passed by cmdline. If passed, that value is used and not touched.

Also, reserved_frac is taken into account on the automatic calculation.

Commit:
Correct message for minimum block count

Must take into account the reserved blocks.

Commit:
Correct also commented code to take into account reserved blocks

xOneca commented:
Although tests pass flawlessly, it seems that there are some blocks that are not being counted in statistics, and thus there aren't enough blocks for some not-so-large folders.

xOneca commented:
I just `make && ./genext2fs -v -m 0 -d . ../aa` the project and it gives:

    genext2fs 1.4.2
    ./genext2fs: couldn't allocate a block (no free space)

I don't say that the flaw is in stats gathering, but I think there must be some blocks that must be taken into account, that aren't. If I remove the `-m 0` it creates the fs without throubles, but it should not be using those reserved blocks, should it?

xOneca committed:
Indirect blocks must be taken into account for stats

Add indirect blocks to file block count stats.

bestouff commented on genext2fs.c:
```diff
@@ -166,8 +166,8 @@ static int blocksize = 1024;
 /* Percentage of blocks that are reserved.*/
-#define RESERVED_BLOCKS       5/100
-#define MAX_RESERVED_BLOCKS  25/100
+#define RESERVED_BLOCKS       (5/100.)
```
I'm not fan of introducing floats just for this.  
Could you please instead use something like
```c
#define TO_RESERVED_BLOCKS(n) ((n) * 5 / 100)
```
?

xOneca committed:
Restore (MAX_)RESERVED_BLOCKS to integer division

Leave the two constants as were defined originally. This reverts the changes
made by commit bfc28c76131695283753084da306e6c73ab8b1ce.
